### ISSUE TITLE
[Self managed]: elastic_agent.metricbeat/filebeat datastreams generated on installing fleet-server agent.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/fleet-server#376]
**Kibana version:** 7.13.0 BC-7 Kibana self managed environment

**Host OS and Browser version:** Windows 10 x64, All

**Build Details:**
```
 Artifact link used: https://staging.elastic.co/7.13.0-8eb98cbf/summary-7.13.0.html
 BUILD: 40864
 COMMIT: 6ce6847436ff9bef0ad91268b6585e0f9339c9fd
```

**Preconditions:**
1. 7.13.0 BC-7 self-managed Kibana environment should be available.
2. Windows 10 x64 Fleet Server agent must be installed using Default Fleet server policy having only Fleet Server integration.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Restart elastic-agent from Services.
3. Navigate to Data Streams tab. 
4. Observe data for few datasets not generating after agent restart. 

**Expected Result:**
Data streams should restart for all datasets on restarting Fleet Server agent.

**Logs:**
[restart issue logs.zip](https://github.com/elastic/fleet-server/files/6508189/restart.issue.logs.zip)

**Note:**
- This issue is observed on self-managed kibana, when first fleet server agent is installed with Default Fleet Server Policy.
- Before restarting elastic-agent data streams for all the datasets were generating at regular intervals. 

**Screenshot:**
![4](https://user-images.githubusercontent.com/77374876/118803046-81739780-b8c0-11eb-9e5c-c17eb3d810a3.JPG)

**Comment by amolnater-qasource** [comment]
@dikshachauhan-qasource  Please review.

**Comment by dikshachauhan-qasource** [comment]
Reviewed and assigned to @EricDavisX

**Comment by EricDavisX** [comment]
@ph @ruflin I'd like to understand this before putting to the backlog - I'd want to know a work-around too.  It actually sounds severe, doesn't it?  that is, if it is a real issue and not otherwise accounted for in other tickets.  

The first step to confirm is, with more time (more than 6 minutes, per the screenshot) would the other datastreams not eventually send data?    I don't see what would be different for elastic_agent.elastic_agent for metrics vs logs, there seems no immediate pattern to the problem.  We should test and confirm that all the same beats (system and monitoring) were re-started on the host.

**Comment by EricDavisX** [comment]
ok, digging into more git /email updates, this seems to exactly relate - metricbeat is crashed on a restart.
https://github.com/elastic/beats/issues/25785

We can close as a dupe if everyone feels confident

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.13.0 BC-9 self-managed Kibana Environment. 
> The first step to confirm is, with more time (more than 6 minutes, per the screenshot) would the other datastreams not eventually send data?

We observed it for more than 30 minutes and still no new data for few datasets was observed.
Please refer below screenshot:
![1](https://user-images.githubusercontent.com/77374876/119305313-3376df00-bc86-11eb-855b-4b7f160603b8.JPG)

**Logs:**
[logs.zip](https://github.com/elastic/fleet-server/files/6530255/logs.zip)

**Build details:**
Build: 40865
Commit: 9863e88bd63ad546b9d36e6b0c0c55cb65dd9081

Please let us know if anything else is required.
Thanks
QAS

**Comment by EricDavisX** [comment]
Adding the 7.13.1 label to help prioritize it until such time as we are confident it is a duplicate and close it, or we otherwise fix / resolve it.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 

Thanks for the feedback on slack @michalpristas 
We have revalidated this issue and observed that on changing the logging level to "debug" we are getting data for all the expected datasets after agent reboot.

Hence closing this out.

Thanks
QAS

**Comment by EricDavisX** [comment]
I will work with Amol to do a retest based on my understanding and we'll report back indeed if anything further can be confirmed as a bug.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.13.0 self managed kibana environment with "info" logging level.
We have observed 15-20 minutes before and after reboot.

Please find below the observation table for the same.

DATASET | Type | Before Reboot | After Reboot
-- | -- | -- | --
elastic_agent.fleet_server | metrics | Generating Normally | Generating Normally
elastic_agent.elastic_agent | metrics | Generating Normally | Generating Normally
elastic_agent.metricbeat | metrics | Generating Normally | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent.filebeat | metrics | Generating Normally | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent.fleet_server | logs | Generated only Once in beginning of Fleet Server agent installation. | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent | logs | Generated only Once in beginning of Fleet Server agent installation. | Generated only Once in beginning of Fleet Server agent reboot.

**Screenshots:**
![DataStreams](https://user-images.githubusercontent.com/77374876/119945027-98964180-bfb2-11eb-8e3c-20eb690baf0d.jpg)

Please let us know if anything else is required.

Thanks
QAS

**Comment by EricDavisX** [comment]
@michalpristas I am re-opening this for a re-review.  It looks to me that after a reboot, the Agent stops collecting dataset elastic_agent.metricbeat and elastic_agent.filebeat data.  Can we discuss / review please?

**Comment by EricDavisX** [comment]
@amolnater-qasource can you re-test this please? we believe this and
https://github.com/elastic/beats/issues/25829 are dupes (or same root cause) and are both fixed now.  thank you

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.14.0 self-managed Kibana and found same observations as shared in comment https://github.com/elastic/fleet-server/issues/376#issuecomment-850217774

**Build details:**
```
Build: 41559
Commit: 9838db392e7fcfc12f004b68fb1b09739f131148
Artifact Link: https://snapshots.elastic.co/7.14.0-28665d9b/downloads/beats/elastic-agent/elastic-agent-7.14.0-SNAPSHOT-windows-x86_64.zip
```

Please let us know if we are missing anything.

Thanks
QAS

**Comment by EricDavisX** [comment]
As of today the 7.14 snapshot is 7 days old, I'm not sure it has the latest fixes in it we need.  I requested the re-testing with hopes that the build was new enough - that is my fault.  Ideally, we'll re-test when the 7.x build is confirmed as new.  Let's wait.

**Comment by EricDavisX** [comment]
all of the builds are green - so we have new artifacts, please retest this and report back.  I am wondering if we still have some issue seen in this and https://github.com/elastic/beats/issues/26034

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.14.0 self managed kibana environment with "info" logging level.
We have observed 15-20 minutes before and after reboot.

Observations are similar to the one's shared earlier at https://github.com/elastic/fleet-server/issues/376#issuecomment-850217774
**Observation table:**

<BODY TEXT="#000000">
<TABLE FRAME=VOID CELLSPACING=0 COLS=4 RULES=NONE BORDER=0>
	<COLGROUP><COL WIDTH=231><COL WIDTH=86><COL WIDTH=206><COL WIDTH=206></COLGROUP>
	<TBODY>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=231 HEIGHT=21 ALIGN=CENTER VALIGN=MIDDLE><B><FONT SIZE=3>Dataset</FONT></B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=86 ALIGN=CENTER VALIGN=MIDDLE><B><FONT SIZE=3>Type</FONT></B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=206 ALIGN=CENTER VALIGN=MIDDLE><B><FONT SIZE=3>Before reboot</FONT></B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=206 ALIGN=CENTER VALIGN=MIDDLE><B><FONT SIZE=3>After reboot</FONT></B></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=20 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent.elastic_agent</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=20 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent.fleet_server</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=38 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent.filebeat</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generated only once after reboot</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=38 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent.metricbeat</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generating regularly</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generated only once after reboot</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=38 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent.fleet_server</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>logs</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generated only Once</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Generated only once after reboot</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=20 ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>elastic_agent</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>logs</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Missing</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT SIZE=3>Missing</FONT></TD>
		</TR>
	</TBODY>
</TABLE>
</BODY>

</HTML>

**Build details:**
```
Build: 42089
Commit: 67a71c75d2da40e49fba2620f488c9b4ce2467d2
Artifact Link: 
https://snapshots.elastic.co/7.14.0-15b00b37/downloads/elasticsearch/elasticsearch-7.14.0-SNAPSHOT-windows-x86_64.zip
https://snapshots.elastic.co/7.14.0-15b00b37/downloads/kibana/kibana-7.14.0-SNAPSHOT-windows-x86_64.zip
https://snapshots.elastic.co/7.14.0-15b00b37/downloads/beats/elastic-agent/elastic-agent-7.14.0-SNAPSHOT-windows-x86_64.zip
```

**Note:**
We have reported issue for missing "elastic_agent" dataset at https://github.com/elastic/beats/issues/26518

Please let us know if anything else is required from our end.
Thanks

**Comment by ph** [comment]
@michalpristas is looking into this one.

**Comment by michalpristas** [comment]
missing dataset most likely duplicate of https://github.com/elastic/beats/issues/26518
i will check missing data on other ones

**Comment by michalpristas** [comment]
do we have any logs from experiment above?
i'm using same version and cannot reproduce. sometimes i needed to hit refresh on UI multiple times for timestamps to get updated though, sometimes they got updated to some stale value and then after few hits to latest ones. maybe some kind of cache (using cloud instance)

**Comment by EricDavisX** [comment]
@amolnater-qasource can you re-run the test and capture the relevant logs (Agent, Fleet Server, Metricbeat, Filebeat) - the last time we posted logs was May 24, I am hoping things may be working better with recent fixes (the one Michal notes was closed out, so that is good).

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this on 7.14.0 BC-2 self-managed Kibana environment.

**Please find the required Logs as follows:**
[logs.zip](https://github.com/elastic/fleet-server/files/6814033/logs.zip)

**Build details:**
```
Build: 42401
Commit: 9826a943dc2e47f26ec6de94816e7d297b752994
Artifact Link: https://staging.elastic.co/7.14.0-e99135ef/summary-7.14.0.html
```

**Screenshot:**
![2](https://user-images.githubusercontent.com/77374876/125574404-a34a33c0-daa2-4082-97b3-7cce73fae713.JPG)

Please let us know if anything else is required from our end.

Thanks
QAS

**Comment by EricDavisX** [comment]
Please do re-test on BC3 or newer snapshot (the BC2 is a week old at his point, and fixes were just merged we think).

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.14.0 BC-3 self managed Kibana environment with "info" logging level.

Observations are still same, please find in below table: 
DATASET | Type | Before Reboot | After Reboot
-- | -- | -- | --
elastic_agent.fleet_server | metrics | Generating Normally | Generating Normally
elastic_agent.elastic_agent | metrics | Generating Normally | Generating Normally
elastic_agent.metricbeat | metrics | Generating Normally | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent.filebeat | metrics | Generating Normally | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent.fleet_server | logs | Generated only Once in beginning of Fleet Server agent installation. | Generated only Once in beginning of Fleet Server agent reboot.
elastic_agent | logs | Generated only Once in beginning of Fleet Server agent installation. | Generated only Once in beginning of Fleet Server agent reboot.

**Build details:**
```
BC-3 Artifact Link: https://staging.elastic.co/7.14.0-682a8012/summary-7.14.0.html
Build: 42545
Commit: c314921a9893e0b46d9a3958f5520e3d6b1ce7d5
```
**Screenshots:**
![4](https://user-images.githubusercontent.com/77374876/125743666-f9237c52-188e-4e85-9712-b5222de0351b.jpg)

Thanks
QAS

**Comment by michalpristas** [comment]
so you don't see no metrics coming in dashboards as well. 

@michel-laterman were you able to find time to see if you can repro locally?

**Comment by michel-laterman** [comment]
I'm having issues recreating this locally; I'm having issues setting up my environment

**Comment by michel-laterman** [comment]
@amolnater-qasource, I'm unable to reproduce this on QA cloud with a locally running agent/fleet server. There have been occasion issues where Kibana fails to refresh the data stream displays, but clicking the "reload" button on the upper right of the UI resolves this.

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
>QA cloud with a locally running agent/fleet server. 

This issue is not reported for Cloud builds. It is only reproducible on self-managed/on-prem setup.

>There have been occasion issues where Kibana fails to refresh the data stream displays, but clicking the "reload" button on the upper right of the UI resolves this.

You can check that we have shared our observations based on nearly 30 mins test at https://github.com/elastic/fleet-server/issues/376#issuecomment-846802757. We were getting new logs for two of the datasets however not for others.

Thanks
QAS

**Comment by michel-laterman** [comment]
I think I have recreated this using `elastic-package stack up` on the latest snapshot (on macos). I can see the same behaviour if I try to restart the docker container. The log messages look the same below is the errors in metricbeat_monitoring-json.log:
```
{"log.level":"info","@timestamp":"2021-07-19T23:07:01.765Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.state: failure to apply state schema: 4 errors: key `management` not found; key `module` not found; key `outpu
t` not found; key `queue` not found","service.name":"metricbeat","event.dataset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-19T23:07:01.771Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.stats: failure to apply stats schema: 1 error: key `libbeat` not found","service.name":"metricbeat","event.dat
aset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-19T23:07:11.765Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.state: failure to apply state schema: 4 errors: key `queue` not found; key `management` not found; key `module
` not found; key `output` not found","service.name":"metricbeat","event.dataset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-19T23:07:11.771Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.stats: failure to apply stats schema: 1 error: key `libbeat` not found","service.name":"metricbeat","event.dat
aset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-19T23:07:21.766Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.state: failure to apply state schema: 4 errors: key `output` not found; key `queue` not found; key `management
` not found; key `module` not found","service.name":"metricbeat","event.dataset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-19T23:07:21.771Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.stats: failure to apply stats schema: 1 error: key `libbeat` not found","service.name":"metricbeat","event.dat
aset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
```
`elastic_agent.elastic_agent` and elastic_agent.fleet_server` metrics appear in the data stream up to date, `elastic_agent.fleet_server` and `elastic_agent` logs timestamps correspond to a restart (I believe because they do not emit log entires when everything is running as expected), and the filebeat and metricbeat metricsets no transmitted. This is probably due to the above errors with metricbeat

**Comment by michalpristas** [comment]
this looks like what metricbeat receives during call to stats does not conform to schema defined in beats module of metricbeat (hence missing fields) 
maybe it's returning empty body or some error message. 
does it go live after restart is performed? which container are you trying to restart?

**Comment by michel-laterman** [comment]
@michalpristas, I was trying to restart the fleet-server container

**Comment by michel-laterman** [comment]
The error logs above are from metricbeat trying to collect state/stats from `fleet-server`.
The state endpoint returns `null` and the stats endpoint is missing the `libbeat` attribute.
However this is the case when the fleet-server initially starts (and all streams work as expected)

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
We have revalidated this issue on 7.14.0 BC-3 self managed Kibana with **"debug"** logging level.

Please find our observations below:
<HTML>
<HEAD>
	
	
</HEAD>

<BODY TEXT="#000000">
<TABLE FRAME=VOID CELLSPACING=0 COLS=3 RULES=NONE BORDER=0>
	<COLGROUP><COL WIDTH=228><COL WIDTH=86><COL WIDTH=213></COLGROUP>
	<TBODY>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=228 HEIGHT=32 ALIGN=CENTER VALIGN=MIDDLE><B>Dataset</B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=86 ALIGN=CENTER VALIGN=MIDDLE><B>Type</B></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" WIDTH=213 ALIGN=CENTER VALIGN=MIDDLE><B><FONT FACE="Times New Roman">Reboot behaviour</FONT></B></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=32 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent.elastic_agent</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generating Normally </FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=32 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent.fleet_server</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generating Normally </FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=32 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent.fleet_server</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">logs</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generating Normally  </FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=62 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">logs</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generating Normally [After longer period of time]</FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=77 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent.metricbeat</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">metrics</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generated only Once in beginning of Fleet Server agent reboot. </FONT></TD>
		</TR>
		<TR>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" HEIGHT=77 ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">elastic_agent.filebeat</FONT></TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE>metrics</TD>
			<TD STYLE="border-top: 1px solid #000000; border-bottom: 1px solid #000000; border-left: 1px solid #000000; border-right: 1px solid #000000" ALIGN=CENTER VALIGN=MIDDLE><FONT FACE="Times New Roman">Generated only Once in beginning of Fleet Server agent reboot. </FONT></TD>
		</TR>
	</TBODY>
</TABLE>
</BODY>

</HTML>

**Screenshot:**
![23](https://user-images.githubusercontent.com/77374876/126432350-94f9027b-cff4-4880-b648-622f3953dec1.JPG)

**Logs:**
[logs.zip](https://github.com/elastic/fleet-server/files/6852872/logs.zip)

cc: @michalpristas 

Please let us know if we are missing anything.
Thanks
QAS

**Comment by michel-laterman** [comment]
Alright, after some discussion I think that the description for the bug is wrong.

The `elastic_agent.metricbeat` and `elastic_agent.filebeat` streams should **not be generated** by the fleet-server default policy. However, currently before the server is restarted the metricbeat_monitor instance is attempting to gather this data. 

If you take a look at the entries from these datasets, it will show errors:
<img width="2525" alt="Screen Shot 2021-07-27 at 10 05 28 AM" src="https://user-images.githubusercontent.com/82832767/127197517-9a11467b-c71f-451f-bb45-c916a97acede.png">

These socket errors are expected as metricbeat/filebeat should not be running integrations (just the `_monitor` versions to gather agent monitoring info).

So it appears as though the bug is that on startup, the metricbeat instance attempts to gather this data. A restart corrects this.

On startup (before a restart) the elastic-agent shows that the fleet-server is re-configuring:
```
bash-4.2$ elastic-agent status
Status: HEALTHY
Message: (no message)
Applications:
  * fleet-server	(CONFIGURING)
    Re-configuring
  * filebeat	(HEALTHY)
    Running
  * metricbeat	(HEALTHY)
    Running
```
The metricbeat_monitor log shows errors connecting to metricbeat and filebeat (same as the metrics stream)
```
{"log.level":"info","@timestamp":"2021-07-27T15:00:13.219Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset http.json: error making http request: Get \"http://unix/stats\": dial unix /usr/share/elastic-agent/data/tmp/defaul
t/filebeat/filebeat.sock: connect: no such file or directory","service.name":"metricbeat","event.dataset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-07-27T15:00:13.219Z","log.origin":{"file.name":"module/wrapper.go","file.line":259},"message":"Error fetching data for metricset beat.stats: error making http request: Get \"http://unix/stats\": dial unix /usr/share/elastic-agent/data/tmp/defau
lt/metricbeat/metricbeat.sock: connect: no such file or directory","service.name":"metricbeat","event.dataset":"metricbeat_monitor-json.log","ecs.version":"1.6.0"}
```

After a restart, fleet-server appears healthy (`elastic-agent status` shows running).

**Comment by EricDavisX** [comment]
Thanks Michael!

@amolnater-qasource @dikshachauhan-qasource if you agree and understand the reasoning, I would like to know.  In this case, too, you can update the short description to align to the 'inverse' issue that is identified.  We can also update our test suite steps to confirm what is configured and note the current bug, until such time as it may be fixed.  Thanks.  I'll mark it as 'done' from the Urgent review side, this is nice to have it off the list even if we still have a lesser priority bug we can evaluate fixing.

It remains on the 7.15 candidate list, and we can evaluate it against other issues / features we want to work on.

**Comment by amolnater-qasource** [comment]
Thanks @michel-laterman for looking up into this issue.
@EricDavisX  yes we agree with michel as on cloud also we observed that only 4 datasets are generated for Fleet Server agent.
No elastic_agent.metricbeat/filebeat data is generated for fleet server on 7.14.0 BC-5 cloud build.

**Screenshot:**
![9](https://user-images.githubusercontent.com/77374876/127607607-4f4fc982-e0bd-4f46-8173-4ed169846df6.JPG)

Further we had a query regarding this: Is there any action, due to which elastic_agent.metricbeat/filebeat could generate in future?

We have updated our Expected result for self managed Fleet server testcase at [C77071](https://elastic.testrail.io/index.php?/cases/view/77071&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=25988)

We will be re-testing it on self-managed 7.15 when it will be fixed.

Please let us know if anything else is required from our end.
Thanks
QAS
